src/tlv/TlvObject.ts

```diff
@@ -80,13 +80,14 @@ export class ObjectSchema<F extends TlvFields> extends TlvSchema<TypeFromFields<
             const { field, name } = fieldName;
             result[name] = field.schema.decodeTlvInternalValue(reader, elementTypeLength);
         }
-        this.validate(result);
         return result as TypeFromFields<F>;
     }
 
     override validate(value: TypeFromFields<F>): void {
         for (const name in this.fieldDefinitions) {
+            if (this.fieldDefinitions[name].optional && (value as any)[name] === undefined) continue;
```

Comment:
nit: you can extract this instead of repeating it.
const fieldValue = (value as any)[name];
const { optional, schema } = this.fieldDefinitions[name];